compiler/src/main/kotlin/io/spine/protodata/renderer/InsertionPoint.kt

```diff
@@ -49,7 +64,84 @@ public interface InsertionPoint {
      * @return the line number at which the insertion point should be added.
      * @see LineNumber
      */
-    public fun locate(lines: List<String>): LineNumber
+    @Deprecated("Use locate(Text) instead.")
+    public fun locate(lines: List<String>): LineNumber {
+        val coords = locate(text(lines))
+        return when (coords.specCase) {
+            WHOLE_LINE -> LineNumber.at(coords.wholeLine)
+            INLINE -> {
+                loggerFor(InsertionPoint::class.java)
+                    .atWarning()
+                    .withStackTrace(FULL)
+                    .log("`locate(List<String>)` does not support inline insertion. " +
+                            "A whole line insertion will be generated.")
+                LineNumber.at(coords.inline.line)
+            }
+            END_OF_FILE -> LineNumber.endOfFile()
+            NOT_IN_FILE -> LineNumber.notInFile()
+            else -> throw IllegalStateException("Unexpected file coordinates `$coords`.")
+        }
+    }
+
+    /**
+     * Locates the site where the insertion point should be added.
+     *
+     * An insertion point should only appear once in a file.
+     *
+     * @param text the existing code
+     * @return the coordinates in the file where the insertion point should be added
+     * @see SourceFile.at
+     * @see SourceFile.atInline
+     */
+    public fun locate(text: Text): FileCoordinates
```

Comment:
It does not feel right. We locate text and get file coordinates. Come on!.. :)

File coordinates feels like drive/directory/.../directory thing.